walk/git_test.go

```diff
@@ -66,3 +70,85 @@ func TestGitReader(t *testing.T) {
 	as.Equal(int32(0), statz.Value(stats.Matched))
 	as.Equal(int32(0), statz.Value(stats.Formatted))
 }
+
+func TestGitStagedReader(t *testing.T) {
+	as := require.New(t)
+
+	tempDir := test.TempExamples(t)
+
+	// init a git repo
+	cmd := exec.Command("git", "init")
+	cmd.Dir = tempDir
+	as.NoError(cmd.Run(), "failed to init git repository")
+
+	// add everything to the git index
+	cmd = exec.Command("git", "add", ".")
+	cmd.Dir = tempDir
+	as.NoError(cmd.Run(), "failed to add everything to the index")
+
+	// commit
+	cmd = exec.Command("git", "commit", "-m", "initial commit")
+	cmd.Dir = tempDir
+	as.NoError(cmd.Run(), "failed to commit")
+
+	// should be empty
```

Comment:
What is the thing that should be empty? Is this referring to the `n` that we assert is 0 a few lines down?

I think I'm getting tripped up because the comment feels "attached" to this block of code that's constructing the `NewGitStagedReader`, and that block of code doesn't seem to be checking about the emptiness of anything.